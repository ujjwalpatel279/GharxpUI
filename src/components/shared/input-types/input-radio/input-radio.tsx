import React, { forwardRef, useEffect, ChangeEvent } from 'react';
import { useForm } from 'react-hook-form';
import { InputListItem } from '..';

interface Options {
  id: string | number;
  value: string;
}

interface InputRadioProps {
  name: string;
  legend: string;
  options: Options[];
  setValue?: ReturnType<typeof useForm>['setValue'];
  value?: string; //option.value
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  disabled?: boolean;
}

type Ref = HTMLInputElement;

export const InputRadio = forwardRef<Ref, InputRadioProps>(
  ({ name, legend, options, setValue, value, onChange, disabled }, ref) => {
    useEffect(() => {
      if (value) {
        setValue(name, value);
      }
    }, [value]);

    return (
      <fieldset>
        <legend>{legend}</legend>
        <div className="l-flex l-flex--wrap">
          {options?.map((option: Options) => (
            <div key={option.id}>
              <InputListItem
                type="radio"
                id={option.id}
                name={name}
                value={option.value}
                ref={ref}
                onChange={onChange}
                disabled={disabled}
              />
            </div>
          ))}
        </div>
      </fieldset>
    );
  },
);
InputRadio.displayName = 'InputRadio';
