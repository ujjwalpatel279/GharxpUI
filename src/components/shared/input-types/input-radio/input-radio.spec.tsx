import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { InputRadio } from './input-radio';

describe('Renders the Radio Group Component', () => {
  afterEach(cleanup);

  it('renders the radio group component', () => {
    const { container } = render(
      <InputRadio name="testName" options={[{ id: '1', value: 'test' }]} legend="Radio legend" ref={() => null} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
