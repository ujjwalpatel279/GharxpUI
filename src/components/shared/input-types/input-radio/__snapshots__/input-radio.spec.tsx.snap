// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Radio Group Component renders the radio group component 1`] = `
<fieldset>
  <legend>
    Radio legend
  </legend>
  <div
    class="l-flex l-flex--wrap"
  >
    <div>
      <label
        class="c-input-list-item"
        for="1"
      >
        <input
          id="1"
          name="testName"
          type="radio"
          value="test"
        />
        <span
          class="c-radio-input__inner"
        >
          <span>
            test
          </span>
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;
